statsmodels/tsa/statespace/regime_switching/kim_filter.py

```diff
@@ -0,0 +1,635 @@
+import numpy as np
+from scipy.misc import logsumexp
+from scipy.optimize import linprog
+from statsmodels.tsa.statespace.kalman_filter import KalmanFilter
+
+try:
+    from scipy.stats import multivariate_normal
+    multivariate_normal_logpdf = multivariate_normal.logpdf
+except ImportError:
+    def multivariate_normal_logpdf(x, mean=None, cov=None):
+        n = x.shape[0]
+        if np.linalg.matrix_rank(cov) < n:
+            raise RuntimeError('Cov matrix is singular.')
+        #TODO: to test this
+        return -0.5 * n * np.log(np.pi) - 0.5 * np.log(np.linalg.det(cov)) - \
```

Comment:
We don't have a specified or strict policy for this. The tendency is that everyone moves to parentheses, but it's not ruled out.
(I often prefer temp variables in my code if there are no natural parenthesis)

I don't like `return` statements that go over several lines. It's better to create to assign to a name and then return it.